File: LSC/Sample/iOS_OSX/Common/LSCTNativeData.m
//
//  LSCTNativeData.m
//  Sample
//
//  Created by admin on 2017/4/17.
//  Copyright © 2017年 hy. All rights reserved.
//

#import "LSCTNativeData.h"

@interface LSCTNativeData ()

@property (nonatomic, strong) NSMutableDictionary *dict;

@end

@implementation LSCTNativeData

- (instancetype)init
{
    if (self = [super init])
    {
        self.dict = [NSMutableDictionary dictionary];
    }
    return self;
}

+ (LSCTNativeData *)createData
{
    return [[LSCTNativeData alloc] init];
}

- (void)setData:(NSString *)data key:(NSString *)key
{
    [self.dict setObject:data forKey:key];
}

- (NSString *)getData:(NSString *)key
{
    return [self.dict objectForKey:key];
}

@end

File: LSC/Sample/iOS_OSX/Sample-OSX-Swift/LSCTNativeData.swift
//
//  LSCTNativeData.swift
//  Sample
//
//  Created by admin on 2017/4/17.
//  Copyright © 2017年 hy. All rights reserved.
//

import Cocoa
import LuaScriptCore_OSX_Swift

@objc(LSCTNativeData)
class LSCTNativeData: NSObject, LuaExportType
{
    @objc var dataId : String? = nil;
    
    private var _data : Dictionary<String, String> = Dictionary();
    
    @objc public class func createData () -> LSCTNativeData
    {
        return LSCTNativeData();
    }
    
    @objc public func setData(value : String, key : String) -> Void
    {
        _data[key] = value;
    }
    
    @objc public func getData(key : String) -> String
    {
        return _data[key]!;
    }
}
